### DISCUSSION TITLE
How to redirect on the server side when req is originating from the browser

### DISCUSSION THREAD

**Discussion by semyou** [vercel/next.js#13032] [Help]
Imagine the case when a user submits their login form. I want to be able to redirect on the api route on the server to a logged-in page. How do I achieve this?
Right now, browser sends a fetch with post to the api router /login. The API returns a res with status 200. I check if response.ok? then I do a router.push('/logged-in')

**Comment by rafinskipg** [comment]
Yeas @semyou  that's the approach.

**Comment by semyou** [comment]
@rafinskipg but that is super inefficient. Why not redirect straight from the API ?

**Reply by rafinskipg** [reply]
@semyou  what do you mean by super inefficient? You can not redirect straight from the API an ajax request. You can redirect straight from the API if you do a normal form submission. But for user experience the correct way is the one described before.

It is not super inefficient. It is being handled on the client-side. It does not cost you computation time.
